Automattic/jetpack
# VideoPress: Allow feature for free sites and add paid product purchase flow

VideoPress is being added as a standalone product to Jetpack! This PR changes the way VideoPress is configured to allow access for free sites (limited to 1 video, enforced on wp.com side), and adds support for purchasing the new VideoPress product from areas of the Jetpack dashboard.

There's a Calypso PR to add the product to the pricing pages and other areas of Jeptack Cloud: https://github.com/Automattic/wp-calypso/pull/56103. _It also adds a nice `Thank You` dialog that is displayed after the purchase. Without that merged, you will just see a Jetpack Free thank you message while testing this PR._

**Screenshots**
_At-a-glance dashoard now has VideoPress card:_
<img width="1026" alt="Screen Shot 2021-09-08 at 4 02 58 PM" src="https://user-images.githubusercontent.com/789137/132591713-51e3eb43-dc7a-46af-b2fb-e25a7bb76378.png">

_Settings->Performance VideoPress card always displays w/ upsell banner:_
<img width="1020" alt="Screen Shot 2021-09-08 at 4 04 23 PM" src="https://user-images.githubusercontent.com/789137/132591875-19c97d1d-f1f0-4666-90c4-b792a73824e7.png">

_Storage Remaining Indicator after upgrading:_
<img width="985" alt="Screen Shot 2021-09-15 at 1 07 41 PM" src="https://user-images.githubusercontent.com/789137/133513464-214b3e6d-8b73-404e-9b11-e78c5a080b37.png">

#### Changes proposed in this Pull Request:
* Enables the VideoPress module for free Jetpack sites.
* Updates `Video hosting` terminology to be `Jetpack VideoPress` or just `VideoPress`.
* Adds upsell banners if a user is on a free site without a plan or the VideoPress upgrade.

#### Jetpack product discussion
See pxWta-Zz-p2

#### Does this pull request change what data or activity we track or use?
I believe clicks on the upgrade buttons are tracked? This isn't new but the `jetpack_videopress` product hasn't been tracked yet.

#### Testing instructions:
* Apply D66710-code to your wp.com sandbox if it hasn't been merged yet, sandbox `public-api.wordpress.com`, and enable `JETPACK__SANDBOX_DOMAIN` on your Jetpack dev site so that it points to your wp.com sandbox. You also may need to enable HTTP access on your WP.com sandbox.
* On your WordPress.com sandbox, enable the feature flag for Jetpack VideoPress in `mu-plugins/videopress/constants.php`: `define( 'VIDEOPRESS_JETPACK_VIDEO_ENABLED', true );`
* Remove all plans/products from your site so that it is considered Jetpack Free.
* Visit `Settings->Performance` or `At-a-glance` dashboard. You should be able to enable the VideoPress module. Once enabled, it should show an upsell for VideoPress product.
* Head to the calypso media library for the site and upload a video. It should be added to VideoPress.
* Now add another video, you should get a message that you need to update your plan.
* Head back to the dashboard and purchase Jetpack VideoPress. 
* You should now see a storage space indicator, and able to upload more videos. The storage indicator should increase as you add files.
* Existing paid plans that support VideoPress (Like Security) should still support VideoPress uploads.
